pkg/utils/array/unique.go

```diff
@@ -0,0 +1,13 @@
+package array
+
+func Unique[T comparable](strSlice []T) []T {
+	keys := make(map[T]bool)
+	list := []T{}
+	for _, entry := range strSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
```

Comment:
Some Go idioms for your reference:
- it's common to use `map[T]struct{}` to represent sets
- usually zero values are use for initial empty slices (`var list []T`)
- the boolean flag of map index is usually named `ok`/`exists`